### DISCUSSION TITLE
Clarification about playlist shortcuts

### DISCUSSION THREAD

**Discussion by sandersantema** [fooyin/fooyin#396] [Q&A]
I see there are multiple shortcuts having something to do with adding tracks to playlists. I cannot figure out however how I'd select some playlist which a shortcut should add the currently playing track to, i.e. select some playlist as "current" and have the shortcuts add to that playlist. Is this possible?

What I couldn't figure out either is the difference between `Send to current playlist` and `Add to current playlist`.

**Comment by ludouzi** [comment]
For the playlist actions:

`Add to current playlist` will append the tracks to the end of the current playlist.
`Send to current playlist` will send the tracks to the current playlist, replacing all existing tracks.

Adding shortcuts to select specific playlists would be possible though difficult to implement correctly, especially with more than 10 playlists. I'll have a think about that one.

**Reply by sandersantema** [reply]
Ah I see.

I don't believe I really understand playlists in Fooyin all to well (I'm thinking of playlists as they are in iTunes or QuodLibet) but I don't think a shortcut for selecting a specific playlist is necessary. Instead I was thinking of a setting so that the same shortcut which adds the track to the current playlist could be used to send it to a playlist of the users choosing.

For anyone interested, at the moment I've solved this in a different way, using a nushell script, it's dbus plugin and xremap to bind it to a global shortcut. Here's the script:
``` nu
#!/usr/bin/env nu
let metadata = dbus get --dest=org.mpris.MediaPlayer2.fooyin /org/mpris/MediaPlayer2 org.mpris.MediaPlayer2.Player Metadata 
let path = $"($metadata | get xesam:url)\n"

let artists = $metadata | get xesam:artist | str join ","
let title = $metadata | get xesam:title

$path | save --append ~/Documents/playlist.m3u
^notify-send "Added" $"($title)\n($artists)"
```
